pipelines/deployer/deployer.yaml

```diff
@@ -731,10 +731,39 @@ jobs:
                   echo "done"
               fi
           fi
+  - task: generate-terraform-var-overrides
+    image: task-toolbox
+    timeout: 40s
+    params:
+      CLUSTER_NAME: ((cluster-name))
+      ACCOUNT_ROLE_ARN: ((account-role-arn))
+      AWS_REGION: eu-west-2
+      AWS_DEFAULT_REGION: eu-west-2
+    config:
+      platform: linux
+      run:
+        path: /bin/bash
+        args:
+        - -euo
+        - pipefail
+        - -c
+        - |
+          echo "assuming aws deployer role..."
+          AWS_CREDS="$(aws-assume-role $ACCOUNT_ROLE_ARN)"
+          eval "${AWS_CREDS}"
+          # Look for ASGs for this cluster that have a single AZ (i.e. worker ASGs) and make a map of their AZs to desired counts
+          export DESIRED_MAP=$(aws autoscaling describe-auto-scaling-groups | jq "$(echo '[.AutoScalingGroups[] | select ([.Tags[] | select (.Key == "kubernetes.io/cluster/$CLUSTER_NAME") | select (.Value == "owned")] | length == 1) | select (.AvailabilityZones | length == 1) | {key: .AvailabilityZones[0], value: .DesiredCapacity}] | from_entries | {desired_workers_per_az_map: . | tostring}' | envsubst)")
```

Comment:
Speaking personally: I'm not sure I'd ever be in a position to understand this if it needed updating. My preference would be to do this in a proper language (e.g. python would do, node would not :trollface:). Splitting this out into multiple lines with appropriate indentations might be sufficient.